### ISSUE TITLE
adv/adp numberOfCells error/discussion

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1381]
```R
data(adv)
adv@metadata$numberOfCells
```
gives 3, but surely it should be 1. So something is wrong for the built-in dataset (see checklist).

Turning now to `adp`, we need to be careful on `numberOfCells`. Does it mean the original number, or the number after possible subsetting by distance? If the latter, then we need to keep it in-sync with `subset,adp-method`, and that seems tricky. And even if it were adjusted in this way it could get messed up because th user may alter the data directly. Maybe `adp[["numberOfCells"]]` should just return `dim(adp[["data"]]$v)`, actually. Comments, devels?

* [ ] see why `data(adv)` is wrong, and fix the underlying code if it caused the problem
* [ ] take into account any advice from @richardsc or @clayton33 on whether to *remove* this from the metadata, and to instead make `[[,adp-method` and `[[,adv-method` simply check the `dim()` of the relevant field and return that.  I lean to this view.

**Comment by dankelley** [comment]
I redid the dataset from scratch, as below, and the `numberOfCells` is not defined by `read.adv`, but rather by `subset,adv-method` and also by `window`. This is a remnant from copying adp code into adv code. I'll fix the code, but will not be rebuilding the data from scratch. Instead, I will hot-fix the data by loading the old and removing `metadata$numberOfCells`. The reason I'm doing that is that I don't want to get into a rats nest of rebuilding files.

```R
m05VectorBeam <- read.adv("/data/archive/sleiwex/2008/moorings/m05/adv/nortek_1943/raw/adv_nortek_1943.vec",
                          from=as.POSIXct("2008-06-25 00:00:00", tz="UTC"), to=as.POSIXct("2008-07-06 00:00:00", tz = "UTC"),
                          latitude=47.87943, longitude=-69.72533)
m05VectorBeam <- retime(m05VectorBeam, a=0.58, b=6.3892e-07, t0=as.POSIXct("2008-07-01 00:00:00", tz="UTC"))
m05VectorBeam <- subset(m05VectorBeam,
                        as.POSIXct("2008-06-25 13:00:00", tz="UTC") <= time & time <= as.POSIXct("2008-07-03 00:50:00", tz="UTC"))
```

**Comment by dankelley** [comment]
Fixed in "develop" branch, commit 88ca6c68805f89d208ca60b8c67e0b4e043decb5
